lib/pbench/test/functional/server/test_put.py

```diff
@@ -70,3 +74,38 @@ def test_index_all(self, server_client: PbenchServerClient, login_user):
                 ), f"Exceeded timeout: state {state}, status {status}"
                 time.sleep(30.0)  # sleep for half a minute
         assert count == len(self.tarballs), "Didn't find all expected datasets"
+
+    def test_upload_again(self, server_client: PbenchServerClient, login_user):
+        """Try to upload a dataset we've already uploaded. This should succeed
+        but with an OK (200) response instead of CREATED (201)
+        """
+        duplicate = self.tarballs[0]
+        response = server_client.upload(duplicate)
+        assert (
+            response.status_code == HTTPStatus.OK
+        ), f"upload failed with {response.status_code}, {response.text}"
```

Comment:
Perhaps, instead of "failed with" it should say "completed with unexpected status, ", since `CREATED` would be a possible return here, and that's not generally regarded as a "failure".

Reply:
Seems like a very trivial distinction in this case, since an alternate success *is* a failure!